PCI is not yet supported by the riscv targets, so change the virtio alias to use
virtio-blk-device instead.

Upstream-Status: Inappropriate
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>

diff --git a/blockdev.c b/blockdev.c
index 81f95d920b..4882d850fa 100644
--- a/blockdev.c
+++ b/blockdev.c
@@ -958,6 +958,8 @@ DriveInfo *drive_new(QemuOpts *all_opts, BlockInterfaceType block_default_type,
                                    &error_abort);
         if (arch_type == QEMU_ARCH_S390X) {
             qemu_opt_set(devopts, "driver", "virtio-blk-ccw", &error_abort);
+        } else if (arch_type == QEMU_ARCH_RISCV) {
+            qemu_opt_set(devopts, "driver", "virtio-blk-device", &error_abort);
         } else {
             qemu_opt_set(devopts, "driver", "virtio-blk-pci", &error_abort);
         }
